lib/vtls/schannel.c

```diff
@@ -1033,8 +1032,6 @@ schannel_recv(struct connectdata *conn, int sockindex,
     infof(data, "schannel: decrypted data buffer: offset %zu length %zu\n",
           connssl->decdata_offset, connssl->decdata_length);
   }
-  else
-    ret = 0;
 
   /* check if the server closed the connection */
   if(ret <= 0 && ( /* special check for Windows 2000 Professional */
```

Comment:
The problem is that now `ret` can never reach `-1` and therefore the second if-condition on `ret < 0 && sspi_status != SEC_E_OK` is invalid. We need to make sure that errors unknown/unhandled sspi_status values/errors are handled there.